src/controllers/legacycontrollersettings.cpp

```diff
@@ -274,3 +291,143 @@ QWidget* LegacyControllerEnumSetting::buildInputWidget(QWidget* pParent) {
 
     return pComboBox;
 }
+
+LegacyControllerColorSetting::LegacyControllerColorSetting(
+        const QDomElement& element)
+        : AbstractLegacyControllerSetting(element) {
+    m_defaultValue = QColor(element.attribute("default"));
+    m_savedValue = m_defaultValue;
+    m_editedValue = m_defaultValue;
```

Comment:
prefer ctor initializer list. they're evaluated in order and you can refer between members. So this works without issues I'm pretty sure.
```suggestion
        const QDomElement& element)
        : AbstractLegacyControllerSetting(element), m_defaultValue(QColor(element.attribute("default")), m_savedValue(m_defaultValue), m_editedValue(m_defaultValue) {
```